cocotb/clock.py

```diff
@@ -58,9 +82,21 @@ def __init__(self, signal, period, units=None):
         self.mcoro = None
 
     @cocotb.coroutine
-    def start(self, cycles=0):
+    def start(self, cycles=None):
+        """Clocking coroutine.  Start driving your clock by forking a call to
+        this.
+
+        Args:
+            cycles (int, optional): Cycle the clock `cycles` number of times.
+                Defaults to ``None`` (cycle the clock forever).
+        """
         t = Timer(self.half_period)
-        while True:
+        if cycles:
```

Comment:
I think you want to explicitly use `cycles is not None` here - if I request 0 cycles, I'd expect to get none, not infinitely many.